### PR TITLE
prepaire to 4.x 

### PR DESCRIPTION
**Pull request by saeideng**

### CODE REVIEW

File: config/bootstrap.php [hunk 1]
```diff
@@ -69,17 +68,11 @@ function ($code, $message, $file, $line, $context = null) use (&$previousHandler
...
-    });
-} else {
-    // Setup dispatch filter
-    $debugBar = new DebugBarFilter(EventManager::instance(), (array)Configure::read('DebugKit'));
-    $debugBar->setup();
-    DispatcherFactory::add($debugBar);
-}
+EventManager::instance()->on('Server.buildMiddleware', function ($event, $queue) use ($service) {
+    $middleware = new DebugKitMiddleware($service);
+    $queue->insertAt(0, $middleware);
```
Comment (markstory) [inline config/bootstrap.php]: We could move this to the Plugin class' `middleware` hook now.

File: src/Plugin.php [hunk 2]
```diff
@@ -14,10 +14,82 @@
...
+        $service = new ToolbarService(EventManager::instance(), (array)Configure::read('DebugKit'));
+        $this->service = $service;
+
+        if (!$service->isEnabled() || php_sapi_name() === 'cli' || php_sapi_name() === 'phpdbg') {
+            return;
+        }
+
+        $this->setDeprecationHandler($service);
+
+        if (!$this->isEnabled('routes')) {
```
Comment (ADmad) [inline src/Plugin.php]: This is not necessary, there's already a routes() hook method which will handle routes file inclusion.

Comment (lorenzo) [reply src/Plugin.php]: Remove it then?

Comment (ADmad) [reply src/Plugin.php]: Yes

**Comment by ADmad** [comment]
`templates/layout/Email/text/default.php` needs to be reanmed to `templates/layout/email/text/default.php`.

**Comment by saeideng** [comment]
>templates/layout/Email/text/default.php needs to be renamed to templates/layout/email/text/default.php.

if you meant in test_app , I did.

, i will open another PR for replacing `Shell` with `Command`

**Pull request merged**